src/any_agent/serving/task_manager.py

```diff
@@ -0,0 +1,170 @@
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from any_agent.logging import logger
+from any_agent.tracing.agent_trace import AgentTrace
+
+if TYPE_CHECKING:
+    from any_agent.serving.config import A2AServingConfig
+
+
+class TaskData:
+    """Data stored for each task."""
+
+    def __init__(self, task_id: str):
+        """Initialize task data.
+
+        Args:
+            task_id: Unique identifier for the task
+
+        """
+        self.task_id = task_id
+        self.agent_trace = AgentTrace()
+        self.last_activity = datetime.now()
+        self.created_at = datetime.now()
+
+    def update_activity(self) -> None:
+        """Update the last activity timestamp."""
+        self.last_activity = datetime.now()
+
+    def is_expired(self, timeout_minutes: int) -> bool:
+        """Check if the task has expired.
+
+        Args:
+            timeout_minutes: Timeout in minutes
+
+        Returns:
+            True if task is expired, False otherwise
+
+        """
+        expiration_time = self.last_activity + timedelta(minutes=timeout_minutes)
+        return datetime.now() > expiration_time
+
+
+class TaskManager:
+    """Manages agent conversation tasks for multi-turn interactions."""
+
+    def __init__(self, config: "A2AServingConfig"):
+        """Initialize the task manager.
+
+        Args:
+            config: Serving configuration containing task settings
+
+        """
+        self.config = config
+        self._tasks: dict[str, TaskData] = {}
+        self._last_cleanup: datetime | None = None
+
+    def _should_run_cleanup(self) -> bool:
```

Comment:
Is this really needed? From a quick look it looks like potentially a premature optimization.
Like, how expensive is to always check if there are tasks and are expired vs always running this (which the increased code that we need to have for it)